extensions/blocks/social-previews/modal.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Social Previews modal component.
+ *
+ * Shows individual previews in modal window.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Modal, TabPanel } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import SocialPreviewsUpgrade from './upgrade';
+import { AVAILABLE_SERVICES } from './constants';
+import { SocialServiceIcon } from '../../shared/icons';
+import { getMediaSourceUrl } from './utils';
+import './modal.scss';
+
+const SocialPreviewsModal = function SocialPreviewsModal( {
+	onClose,
+	showUpgradeNudge,
+	image,
+	title,
+	description,
+	url,
+	author,
+} ) {
+	// Inject the service icon into the title
+	const tabs = AVAILABLE_SERVICES.map( service => {
+		return {
+			...service,
+			title: (
+				<Fragment>
+					<SocialServiceIcon serviceName={ service.icon } />
+					{ service.title }
+				</Fragment>
+			),
+		};
+	} );
+
+	return (
+		<Modal
+			onRequestClose={ onClose }
+			title={ __( 'Social Previews', 'jetpack' ) }
+			className="jetpack-social-previews__modal"
+		>
+			{ showUpgradeNudge ? (
+				<SocialPreviewsUpgrade />
+			) : (
+				<TabPanel className="jetpack-social-previews__modal-previews" tabs={ tabs }>
```

Comment:
There's an `orientation="vertical"` prop that would make the arrow key interactions use the Up/Down arrow instead of the Left/Right arrow keys to move between tabs. This may mean we can remove some styles to make it vertical orientation as well.

Reply:
Good spot, addressed this issue just now.

Reply:
I guess it would _maybe_ need to be changed to horizontal based on screen size, but I think the odds that you're using a keyboard on a < 782px screen is a lot lower than on a larger one. That said, it would be nice to be able to update the orientation based on screen size in a follow-up PR. I think setting it to `vertical` is best for now.